File: Java/LeetCode/Largest Number At Least Twice of Others.java
// https://leetcode.com/problems/largest-number-at-least-twice-of-others/

class Solution {

    public int dominantIndex(int[] nums) {
        int max = Arrays.stream(nums).max().getAsInt();
        int maxIndex = -1;

        for (int i = 0; i < nums.length; ++i) {
            if (nums[i] == max) {
                maxIndex = i;
            } else if (nums[i]*2 > max) {
                return -1;
            } 
        }

        return maxIndex;
    }

}

File: Kotlin/LeetCode/Largest Number At Least Twice of Others.kt
// https://leetcode.com/problems/largest-number-at-least-twice-of-others/

class Solution {

    fun dominantIndex(nums: IntArray): Int {
        val max = nums.max()
        var maxIndex = -1

        for (i in 0 until nums.size) {
            if (nums[i] == max) {
                maxIndex = i
            } else if (nums[i]*2 > max) {
                return -1
            } 
        }

        return maxIndex
    }

}
